cartography/intel/opal/__init__.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import neo4j
+import opal_security as opal
+
+from cartography.config import Config
+from cartography.intel.opal.opal_aws import sync_opal_aws
+from cartography.intel.opal.opal_okta import sync_opal_okta
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+PAGE_SIZE = 200
+logger = logging.getLogger(__name__)
+
+
+def _resource_client(configuration: opal.Configuration) -> opal.api.ResourcesApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.ResourcesApi(api_client)
+
+
+def _group_client(configuration: opal.Configuration) -> opal.api.GroupsApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.GroupsApi(api_client)
+
+
+def _owners_client(configuration: opal.Configuration) -> opal.api.OwnersApi:
+    api_client = opal.ApiClient(configuration)
+    return opal.OwnersApi(api_client)
+
+
+def parse_group(group: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Parse a group dict and add the resource_type field.
+    """
+    group["resource_id"] = group.get("group_id")
+    group["resource_type"] = "OPAL_GROUP"
+    group["remote_resource_name"] = group.get("remote_name")
+    return group
+
+
+def get_all_groups(client: opal.api.GroupsApi) -> List[Dict[str, Any]]:
```

Comment:
Can we move this to opal_common.py? I'm thinking this file should pretty much only have the start_* and other orchestration functions.